packages/components/src/components/ScalarIcon/icons/index.ts

```diff
@@ -1,16 +1,21 @@
+import { defineAsyncComponent } from 'vue'
+
 import { type ICONS } from './icons'
 
-const icons = import.meta.glob('./*.svg', { as: 'raw', eager: true })
+const icons = import.meta.glob<SVGElement>('./*.svg')
 
+/**
+ * Generate a vue component from the icon SVGs
+ */
 export const getIcon = (name: string) => {
   const filename = `./${name}.svg`
 
-  if (icons[filename] === undefined) {
+  if (!icons[filename]) {
     console.warn(`Could not find icon: ${name}`)
-    return ''
+    return null
   }
 
-  return icons[filename]
+  return defineAsyncComponent(icons[filename]!)
```

Comment:
So this just transforms the raw SVG files into Vue components? 👀

Reply:
yep I'm assuming thats what we wanted? We can keep em as svgs as well

Reply:
This seems cool, only thing we have to watch out for it wether or not the SVGs added to the folder are well formed (e.g. have correctly set viewboxes and use `currentColor` for the color)

Reply:
I dumped them in from flowIcon, but only the ones that didn't already exist. So the previous ones should keep their currentColor. The new ones might not have it. Should I change em all?